Sources/Networking/HTTPResponse.swift

```diff
@@ -14,19 +14,44 @@
 
 import Foundation
 
-struct HTTPResponse {
+struct HTTPResponse<Body: HTTPResponseBody> {
 
-    typealias Body = [String: Any]
+    typealias Result = Swift.Result<Self, Error>
```

Comment:
This `typealias` is used everywhere. This will allow replacing `Error` with a more specific `Error` so we can forward known typed errors, and make sure that we don't end up with the wrong error type, or with a very complex error hierarchy and the details buried in `underlyingError`s.